datafusion/core/src/datasource/physical_plan/json.rs

```diff
@@ -282,14 +283,15 @@ pub async fn plan_to_json(
                 let mut writer = json::LineDelimitedWriter::new(buffer);
                 writer.write(&batch)?;
                 buffer = writer.into_inner();
-                multipart_writer.write_all(&buffer).await?;
+                multipart_writer.write_all(&buffer).await.map_err(|e| {
```

Comment:
basically the point of this PR is to annotate callsites like this with information about where they were writing to